Partial backport of extern declaration fixes

Partial backport of upstream fixes in commit 30a23dd.  Not all
changes from the commit apply to the older version AGL is currently
using.  Updating to a newer version is complicated by the other
patches AGL applies.

Upstream-Status: Backport [https://github.com/Arkq/bluez-alsa/commit/30a23dd7ce9b00c702fa48545f6b03038f9e17c6]
Signed-off-by: Scott Murray <scott.murray@konsulko.com>

---
diff --git a/src/bluealsa-iface.h b/src/bluealsa-iface.h
index 262fb76..1d3366d 100644
--- a/src/bluealsa-iface.h
+++ b/src/bluealsa-iface.h
@@ -32,8 +32,8 @@
 #define BLUEALSA_RFCOMM_MODE_HSP_AG "HSP-AG"
 #define BLUEALSA_RFCOMM_MODE_HSP_HS "HSP-HS"
 
-const GDBusInterfaceInfo bluealsa_iface_manager;
-const GDBusInterfaceInfo bluealsa_iface_pcm;
-const GDBusInterfaceInfo bluealsa_iface_rfcomm;
+extern const GDBusInterfaceInfo bluealsa_iface_manager;
+extern const GDBusInterfaceInfo bluealsa_iface_pcm;
+extern const GDBusInterfaceInfo bluealsa_iface_rfcomm;
 
 #endif
diff --git a/src/bluez-a2dp.h b/src/bluez-a2dp.h
index 7c0e33b..4a695d9 100644
--- a/src/bluez-a2dp.h
+++ b/src/bluez-a2dp.h
@@ -1,6 +1,6 @@
 /*
  * BlueALSA - bluez-a2dp.h
- * Copyright (c) 2016-2019 Arkadiusz Bokowy
+ * Copyright (c) 2016-2020 Arkadiusz Bokowy
  *
  * This file is a part of bluez-alsa.
  *
@@ -58,6 +58,6 @@ struct bluez_a2dp_codec {
 };
 
 /* NULL-terminated list of available A2DP codecs */
-const struct bluez_a2dp_codec **bluez_a2dp_codecs;
+extern const struct bluez_a2dp_codec **bluez_a2dp_codecs;
 
 #endif
diff --git a/src/bluez-iface.h b/src/bluez-iface.h
index 34269c5..ca8fba5 100644
--- a/src/bluez-iface.h
+++ b/src/bluez-iface.h
@@ -1,6 +1,6 @@
 /*
  * BlueALSA - bluez-iface.h
- * Copyright (c) 2016-2019 Arkadiusz Bokowy
+ * Copyright (c) 2016-2020 Arkadiusz Bokowy
  *
  * This file is a part of bluez-alsa.
  *
@@ -27,7 +27,7 @@
 #define BLUEZ_TRANSPORT_STATE_PENDING "pending"
 #define BLUEZ_TRANSPORT_STATE_ACTIVE  "active"
 
-const GDBusInterfaceInfo bluez_iface_endpoint;
-const GDBusInterfaceInfo bluez_iface_profile;
+extern const GDBusInterfaceInfo bluez_iface_endpoint;
+extern const GDBusInterfaceInfo bluez_iface_profile;
 
 #endif
diff --git a/src/ofono-iface.h b/src/ofono-iface.h
index 4a5f346..00f856b 100644
--- a/src/ofono-iface.h
+++ b/src/ofono-iface.h
@@ -25,6 +25,6 @@
 #define OFONO_AUDIO_CODEC_CVSD 0x01
 #define OFONO_AUDIO_CODEC_MSBC 0x02
 
-const GDBusInterfaceInfo ofono_iface_hf_audio_agent;
+extern const GDBusInterfaceInfo ofono_iface_hf_audio_agent;
 
 #endif
diff --git a/test/server-mock.c b/test/server-mock.c
index 4d4ca78..ecf4664 100644
--- a/test/server-mock.c
+++ b/test/server-mock.c
@@ -32,6 +32,7 @@
 #include "../src/ba-adapter.c"
 #include "../src/ba-device.c"
 #include "../src/ba-transport.c"
+#include "../src/bluez-a2dp.c"
 #include "../src/io.h"
 #define io_thread_a2dp_sink_sbc _io_thread_a2dp_sink_sbc
 #include "../src/io.c"
